src/app.js

```diff
@@ -16,72 +16,69 @@ import {
   help,
   helpSpecific,
   emoji,
-} from './commands';
-import { insertAccount } from './queries';
+} from "./commands";
+import { insertAccount } from "./queries";
 
 // Create an instance of Discord client
 const client = new Discord.Client();
 // Prefix that every command should start with
-const prefix = '!';
+const prefix = "!";
 
 // Bot will only start reacting to information once ready is emitted
-client.on('ready', () => {
+client.on("ready", () => {
   console.log(`Logged in as ${client.user.tag}!`);
 });
 
 // Create an event listener for messages
-client.on('message', msg => {
+client.on("message", msg => {
   const { content, author, channel, member } = msg;
 
-  // Returns true if the message content matches any of
-  // the given commands
-  const commandMatches = commands => {
-    if (Array.isArray(commands)) {
-      for (let i = 0; i < commands.length; i++) {
-        if (`${prefix}${commands[i]}` === content.split(' ')[0]) return true;
-      }
-      return false;
-    }
+  // Ignore messages from the bot
+  if (author.bot) return;
 
-    return `${prefix}${commands}` === content;
-  };
+  //Ignore any message that does not start with prefix
+  if (!content.startsWith(prefix)) return;
 
-  // Do not evaluate a message not starting with  our prefix
-  if (content.startsWith(prefix)) {
-    switch (true) {
-      case commandMatches('ping'):
-        channel.send('pong!');
-        break;
-      case commandMatches('knowme') && !author.bot:
-        knowme(msg);
-        break;
-      case commandMatches('repopulate') &&
-        member.roles.has('363153451833753600'):
-        repopulate(msg);
-        break;
-      case commandMatches(['marshmallow', 'mm']):
-        marshmallow(msg);
-        break;
-      case commandMatches('mine'):
-        mine(msg);
-        break;
-      case content.match(/^(!help|!h) !?[a-zA-Z]+$/) !== null:
-        helpSpecific(msg);
-        break;
-      case commandMatches(['help', 'h']):
-        help(msg);
-        break;
-      case commandMatches(['emoji']):
-        emoji(msg);
-        break;
-      default:
-        break;
-    }
+  const args = content
+    .slice(prefix.length)
+    .trim()
+    .split(/ +/g);
+  const command = args.shift().toLowerCase();
+
+  switch (command) {
+    case "ping":
+      channel.send("pong!");
+      break;
+    case "knowme":
+      knowme(msg);
+      break;
+    case "repopulate":
+      if (member.roles.has("363153451833753600")) repopulate(msg);
+      break;
+    case "marshmallow":
+    case "mm":
+      marshmallow(msg);
+      break;
+    case "mine":
+      mine(msg);
+      break;
+    case "help":
+    case "h":
+      if (args.length === 0) help(msg);
+      else if (args[0].match(new RegExp(`^${prefix}?\\w+`, "g")))
+        helpSpecific(msg, args[0].slice(prefix.length).trim());
+      else helpSpecific(msg, "");
```

Comment:
Re-commenting on newer version: 
1. `//w` = `w`: https://regexr.com/3rqmq
2. `!help` should not accept anything other than `!mine`. It should error out for anything else, wheter it be `mine` or `+mine` or `mine a bced`